### PR TITLE
Spotlight prediction market

### PR DESCRIPTION
**Pull request by Metizik**
<h2>General</h2>
Closes UI part of #120 

Implemented all stated UI for tables with prediction markets overview and market details (opinions related to market). 
Implemented fetching of markets and opinions from blockchain.
Implemented modals for PM creation, Opinion adding, PM resolution.
Implemented filter features for markets and opinions tables.

Left to do / clarify:
- [ ] Use action to create prediction market
- [ ] Use action to buy/sell PM assets (submit opinion)
- [ ] Use action to resolve PM
- [ ] Use action to cancel owned opinions
- [ ] Fix unstable issues with opinions fetching from MarketsStore
- [ ] Improve filter features

UX issues to discuss:
- [ ] yes/no buttons seems like does no effect from UX point of view
- [ ] resolution date (expiry) is not present in most prediction markets and not reflected in UI
- [ ] It is not clear what does opinions table contain
- [ ] Opinions table is out of screen, when there are PMs with long description
- [ ] Resolved PMs are positionaly displayed in overview table

<h2>General</h2>
Please make sure the following is done:

- Pull request is onto develop
- If you haven't already, have a look at
  - https://github.com/bitshares/bitshares-ui/blob/develop/CODE_OF_CONDUCT.md
  - https://github.com/bitshares/bitshares-ui/blob/develop/CONTRIBUTING.md

<h2>Code Preparation</h2>

_Please review all your changes one last time before committing_

- [ ] Check for unused code
- [ ] No unrelated changes are included
- [ ] None of the changed files are reformatting only
- [ ] Code is self explanatory or documented
- [ ] All written text is properly translated (english language)

<h2>Testing</h2>

_The branch has been tested on the following browsers (desktop and mobile view)_

- [ ] Chrome 
- [ ] Opera
- [ ] Firefox
- [ ] Safari

<h2>User interface changes</h2>

_Delete this section if there weren't any UI changes. Please make sure you tested your changes in all themes_

- [ ] Dark
- [ ] Light
- [ ] Midnight

_Please provide screenshots/licecap of your changes below_

### CODE REVIEW

File: app/components/PredictionMarkets/AddOpinionModal.jsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                        </Form.Item>
+                    </Form>
+                </div>
+            </Modal>
+        );
+    }
+}
+
+AddOpinionModal.propTypes = {
+    show: PropTypes.bool,
```
Comment (sschiessl-bcp) [inline app/components/PredictionMarkets/AddOpinionModal.jsx]: please stick to commonly used wording `visible`

File: app/components/PredictionMarkets/PMAssetsContainer.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import AltContainer from "alt-container";
+import PredictionMarkets from "./PredictionMarkets";
+import BindToChainState from "../Utility/BindToChainState";
+
+class PMAssetsContainer extends React.Component {
+    render() {
+        return (
+            <AltContainer
+                stores={[AssetStore, AccountStore, MarketsStore]}
+                inject={{
```
Comment (sschiessl-bcp) [inline app/components/PredictionMarkets/PMAssetsContainer.jsx]: please restrict to the ones actually in use

File: app/components/PredictionMarkets/PredictionMarketDetailsTable.jsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+        let i = 0;
+        filteredOpinions = filteredOpinions.map(item => ({
+            ...item,
+            key: `${item.order_id}${i++}`
+        }));
+
+        return (
+            <div style={{paddingTop: "50px"}} key="overview-table">
+                <Table
```
Comment (sschiessl-bcp) [inline app/components/PredictionMarkets/PredictionMarketDetailsTable.jsx]: use the paginatedlist component to streamline UX

File: app/components/PredictionMarkets/PredictionMarkets.jsx [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+            preselectedAmount: 0,
+            isCreateMarketModalOpen: false,
+            isAddOpinionModalOpen: false,
+            isResolveModalOpen: false
+        };
+        this._updateAssetsList("A");
+    }
+
+    componentWillReceiveProps(np) {
+        if (np.currentAccount !== this.props.currentAccount) {
```
Comment (sschiessl-bcp) [inline app/components/PredictionMarkets/PredictionMarkets.jsx]: can you add some comments please what is done for what purpose?

File: app/components/PredictionMarkets/PredictionMarkets.jsx [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+        if (
+            !this.state.lastAssetSymbol ||
+            this.state.lastAssetSymbol !== searchAsset
+        ) {
+            this._updateAssetsList(searchAsset);
+            this.setState({
+                lastAssetSymbol: searchAsset
+            });
+        }
+        console.log(this.props.markets);
```
Comment (sschiessl-bcp) [inline app/components/PredictionMarkets/PredictionMarkets.jsx]: remove debug logs

File: app/components/PredictionMarkets/PredictionMarkets.jsx [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+        }));
+        this.setState({opinions: [...asks, ...bids]});
+    }
+
+    async _updateAssetsList(lastAsset) {
+        AssetActions.getAssetList.defer(lastAsset, 100);
+    }
+
+    updateMarketsList() {
+        const markets = this.state.assets.map(item => ({
```
Comment (sschiessl-bcp) [inline app/components/PredictionMarkets/PredictionMarkets.jsx]: did not check if that is done, but the markets can be restricted to: PREDICTIONASSET / ITSCOLLATERAL. For exmaple: If SOMEPREDICTION is using BTS as collateral, only that market needs to be shown.

Keep the list based approach nonetheless for future expansion

**Review by sschiessl-bcp — CHANGES_REQUESTED** [review]

File: app/components/PredictionMarkets/PredictionMarkets.jsx (hunk 6 above)
Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/PredictionMarkets.jsx]: Could you describe in more detail, it is not entirely clear, what restriction do you mean?

Comment (sschiessl-bcp) [reply app/components/PredictionMarkets/PredictionMarkets.jsx]: It's only necessary to subscripte to the PREDICTIONASSET / ITSCOLLATERAL market.

Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/PredictionMarkets.jsx]: In this function updateMarketsList() we form the list of prediction markets from list of assets to be shown in UI.
Elsewhere in the code we take all the assets and filter them checking the flag "bitasset_data.is_prediction_market".
Do we need any other filters for prediction markets?
As far as we understand, in expression PREDICTIONASSET / ITSCOLLATERAL predictionasset is the symbol of prediction market, and itscollateral is its backing asset. Is it correct?

Comment (sschiessl-bcp) [reply app/components/PredictionMarkets/PredictionMarkets.jsx]: Yes, that is correct. My comment refers to making sure that we `ONLY` subscribe to the market of the currently selected prediction market asset, and not all.

File: app/components/PredictionMarkets/PMAssetsContainer.jsx (hunk 2 above)
Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/PMAssetsContainer.jsx]: Done

File: app/components/PredictionMarkets/PredictionMarketDetailsTable.jsx (hunk 3 above)
Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/PredictionMarketDetailsTable.jsx]: Done

File: app/components/PredictionMarkets/PredictionMarkets.jsx (hunk 4 above)
Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/PredictionMarkets.jsx]: Splited into several functions

File: app/components/PredictionMarkets/PredictionMarkets.jsx (hunk 5 above)
Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/PredictionMarkets.jsx]: Done

File: app/components/PredictionMarkets/PredictionMarkets.jsx (hunk 6 above)
Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/PredictionMarkets.jsx]: The code has the following logic: we get all the assets from the Chain, filter the list to get only the prediction markets and show the list in the Prediction Markets table. After clicking on one of the rows, we subscribe to the selected market and get the list of limit orders, then we show it in the Opinions table. So seems it's done.

File: app/components/PredictionMarkets/AddOpinionModal.jsx (hunk 1 above)
Comment (VerevkinAlexander) [reply app/components/PredictionMarkets/AddOpinionModal.jsx]: Done

**Comment by sschiessl-bcp** [comment]
Forwarding this for testing

**Pull request closed without merging**